### PR TITLE
docs: EVG-19251 task trace doc fixes

### PR DESCRIPTION
**Pull request by ybrill**
[EVG-19251](https://jira.mongodb.org/browse/EVG-19251)

### Description
User testing showed that the two changes from this PR are necessary for test spans to hook into their task's trace.
Specifically:
~- specify a `service.name` that's the same as the task trace's~
- set the `FlagsSampled` trace flag so it isn't dropped along the way

### CODE REVIEW

File: docs/Project-Configuration/Task_Traces.md [hunk 1]
```diff
@@ -27,11 +27,23 @@ The following examples illustrate how to inject the trace/span IDs into the cont
...
 from opentelemetry import trace, context
-from opentelemetry.trace import NonRecordingSpan, SpanContext
-
+from opentelemetry.trace import NonRecordingSpan, SpanContext, TraceFlags
+from opentelemetry.sdk.resources import SERVICE_NAME, Resource
+from opentelemetry.sdk.trace import TracerProvider
+
+# Task traces in Honeycomb are sent with an "evergreen-agent" service name.
+# We need to send test spans with the same service name since the service name
+# determines the Honeycomb dataset and traces can not span multiple datasets.
```
Comment (opsnlops) [inline docs/Project-Configuration/Task_Traces.md:36]: Is this accurate? I thought it could span datasets, but not environments?

Comment (ybrill) [reply docs/Project-Configuration/Task_Traces.md]: 😅 .
I pushed a change to just make the trace flags change instead.

**Review by opsnlops — APPROVED** [review]
😃

File: docs/Project-Configuration/Task_Traces.md [hunk 2]
```diff
@@ -27,11 +27,12 @@ The following examples illustrate how to inject the trace/span IDs into the cont
...
 ```python
 from opentelemetry import trace, context
-from opentelemetry.trace import NonRecordingSpan, SpanContext
+from opentelemetry.trace import NonRecordingSpan, SpanContext, TraceFlags
 
 span_context = SpanContext(
     trace_id = int("${otel_trace_id}", 16),
-    span_id = int(("${otel_parent_id}", 16)
+    span_id = int(("${otel_parent_id}", 16),
+    trace_flags = TraceFlags(0x01)
```
Comment (IamXander) [inline docs/Project-Configuration/Task_Traces.md:35]: There is an `is_remote` flag that is required here. Could you give guidance on what the correct option should be?

**Pull request merged**